File: projects/fullstack-media-converter/apps/ios/ConvertMate/Engine/ConversionEngine.swift
// MARK: - ConversionEngine.swift
// Platform-agnostic protocol — same concept as packages/core ConversionEngine.

import Foundation

protocol ConversionEngine {
    func canConvert(from inputFormat: String, to outputFormat: String) -> Bool
    func convert(
        job: ConversionJob,
        options: ConversionOptions,
        progress: @escaping (Double) -> Void
    ) async throws -> URL
}

File: projects/fullstack-media-converter/apps/android/app/src/main/java/com/convertmate/conversion/ConversionEngine.kt
package com.convertmate.conversion

import android.content.Context
import android.net.Uri
import com.convertmate.model.ConversionJob
import com.convertmate.model.ConversionOptions
import com.convertmate.model.JobStatus

/**
 * Platform-agnostic interface — same concept as packages/core ConversionEngine.
 * Implementations: ImageConversionEngine, VideoConversionEngine.
 */
interface ConversionEngine {
    fun canConvert(inputFormat: String, outputFormat: String): Boolean
    suspend fun convert(
        context: Context,
        job: ConversionJob,
        options: ConversionOptions,
        onProgress: (Int) -> Unit,
    ): ConversionJob
}
